owltorrent-core/src/main/java/edu/rice/owltorrent/core/serialization/TempParser.java

```diff
@@ -2,19 +2,104 @@
 
 import com.dampcake.bencode.BencodeInputStream;
 import edu.rice.owltorrent.common.entity.Torrent;
+import lombok.NonNull;
+
 import java.io.ByteArrayInputStream;
 import java.io.File;
 import java.io.IOException;
 import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
 
+/**
+ * Torrent file parser.
+ *
+ * @author bhaveshshah
+ */
 public class TempParser {
 
```

Comment:
Can you create a newly named class, for example BenCodeParser instead?

Reply:
Renamed class.